nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -715,6 +785,40 @@ public static Nullness getGenericReturnNullnessAtInvocation(
       MethodInvocationTree tree,
       VisitorState state,
       Config config) {
+
+    List<? extends Tree> typeArgumentTrees = tree.getTypeArguments();
+    com.sun.tools.javac.util.List<Type> explicitTypeArgs =
+        convertTreesToTypes(typeArgumentTrees); // Convert to Type objects
+
+    Type methodType = invokedMethodSymbol.type;
+    Type substitutedReturnType = null;
+    if (methodType instanceof Type.ForAll) {
+      Type.ForAll forAllType = (Type.ForAll) methodType;
+
+      // Extract the underlying MethodType (the actual signature)
+      Type.MethodType methodTypeInsideForAll = (Type.MethodType) forAllType.qtype;
+
+      // Substitute the argument and return types within the MethodType
+      substitutedReturnType =
+          state
+              .getTypes()
+              .subst(methodTypeInsideForAll.restype, forAllType.tvars, explicitTypeArgs);
+    } else {
+      // If it's not a ForAll type, handle it as a normal MethodType
+      Type.MethodType methodTypeElse = (Type.MethodType) invokedMethodSymbol.type;
+      substitutedReturnType =
+          state
+              .getTypes()
+              .subst(
+                  methodTypeElse.restype,
+                  invokedMethodSymbol.type.getTypeArguments(),
+                  explicitTypeArgs);
```

Comment:
I feel like if we only apply the new logic to generic methods, we will always get a `ForAll` type and won't need this `else` case